ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/useReportFormValues.tsx

```diff
@@ -145,7 +145,21 @@ const validationSchema = yup.object().shape({
         })
         .notRequired(),
     schedule: yup.object().shape({
-        intervalType: yup.string().oneOf(intervalTypes).nullable(),
+        intervalType: yup
+            .string()
+            .oneOf(intervalTypes)
+            .nullable()
+            .strict()
+            .test('schedule-is-required', 'A schedule is required', (value, context) => {
+                if (!context.from) {
+                    return false;
+                }
+                const contextValue: ReportFormValues = context.from[context.from.length - 1].value;
```

Comment:
For your information, if you ever need to retrieve the value of the highest ancestor, which essentially means the entire form value, you can utilize `context.from[context.from.length - 1].value`. There is no need to hardcode an index value. Since the depth of nesting in your form object can vary, the index value would differ accordingly. Using `context.from.length - 1` guarantees that you will always obtain the correct value.

Reply:
I was thinking about if it would be useful to create some sort of util function to make using `test` easier. Not sure it's necessary now, but in the future if anyone needs to do other complex validation using the `test` function, we can consider if some util function may be of use here